src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -516,6 +628,44 @@ DataModel::CommandEntry CodegenDataModelProvider::NextAcceptedCommand(const Conc
 
 std::optional<DataModel::CommandInfo> CodegenDataModelProvider::GetAcceptedCommandInfo(const ConcreteCommandPath & path)
 {
+    // Command handler interface MAY override lists of commands
+    CommandHandlerInterface * handler_interface =
+        CommandHandlerInterfaceRegistry::Instance().GetCommandHandler(path.mEndpointId, path.mClusterId);
+
+    if (handler_interface != nullptr)
+    {
+        struct Context
+        {
+            bool commandExists;
+            CommandId targetCommand;
+        } context{ false, path.mCommandId };
+
+        // if a list is available, try to see if the command exists
+        CHIP_ERROR err = handler_interface->EnumerateAcceptedCommands(
+            path,
+            [](CommandId command, void * closure) -> Loop {
+                auto * ctx = static_cast<Context *>(closure);
+                if (ctx->targetCommand == command)
+                {
+                    ctx->commandExists = true;
+                    return Loop::Break;
+                }
+                return Loop::Continue;
+            },
+            &context);
+
+        if (err != CHIP_ERROR_NOT_IMPLEMENTED)
+        {
+            // TODO: we know the command exists, but don't actually have info such as flags. Ideally CHI should
```

Comment:
nit what is CHI? Do you mean CHIP? Even then I don't follow this comment or TODO
```suggestion
            // TODO: we know the command exists, but don't actually have info such as flags. Ideally CHIP should
```

Reply:
I meant `CommandHandlerInterface` ... agree it is confusing. Will try to spell it out.